lib/pbench/agent/tool_data_sink.py

```diff
@@ -186,6 +190,408 @@ def terminate(self):
+                )
+
+            try:
+                os.mkdir(os.path.join(".temp", host))
+                self.logger.debug(
+                    "Created folder :{}".format(os.path.join(".temp", host))
+                )
+            except Exception:
+                self.logger.debug(
+                    "Folder :{} already exists".format(os.path.join(".temp", host))
+                )
+
+            try:
+                for tool in self.host_tools_dict[host]:
+
+                    try:
+                        string = ""
+                        string += "#pmlogconf-setup 2.0\n"
+                        string += "ident   " + self.mapping[tool]["ident"] + "\n"
+                        string += (
+                            "probe   "
+                            + self.mapping[tool]["probe"]
+                            + " ? include : exclude\n\n"
+                        )
+                        for _, metric in enumerate(self.mapping[tool]["metrics"]):
+                            string += "   " + metric + "\n"
+
+                        self.logger.debug(
+                            "%s.summary output for host %s :\n%s", host, host, string
+                        )
+                        with open(
+                            os.path.join(".temp", host, tool + ".summary"), "w"
+                        ) as file:
+                            file.write(string)
+                    except KeyError:
+                        self.logger.error("tool:%s does not exist in mapping", tool)
+
+                # code to build pmlogger configuration file
+                self.logger.info("building pmlogger configuration file")
+                command = "pmlogconf -c -h {} -d {} {}.config ".format(
+                    host, os.path.join(".temp", host), host
+                )
+                self.logger.debug("pmlogconf command:%s", command)
+                out = subprocess.Popen(
+                    [command],
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.STDOUT,
+                    shell=True,
+                )
+                stdout, stderr = out.communicate()
+                self.logger.debug("stdout output:%s", stdout.decode())
+
+                if (
+                    stdout.decode().strip()
+                    != f'Creating config file "{host}.config" using default settings ...\n.'
+                ):
+                    # pmlogger failed to stop. Log it
+                    self.logger.error("not able to build pmlogconf file")
+                else:
+                    self.logger.info("built config file for host:{}".format(host))
+
+                self.pmlog_config_files.append("{}.config".format(host))
+            except Exception as e:
+                # log exception as we are not able to stop logger
+                self.logger.error(
+                    "not able to create a logger conf file\n", e, exc_info=True
```

Comment:
`exc_info=True` causes the logger to add exception information, so I think the `e` argument is unused since there's no substitution placeholder in the message string.

This pattern is repeated several times below.